src/tests/unit/auto/auto_ctput_test.cpp

```diff
@@ -0,0 +1,246 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <common_test_utils/test_constants.hpp>
+#include <ngraph_functions/subgraph_builders.hpp>
+
+#include "cpp/ie_plugin.hpp"
+#include "plugin/mock_load_network_properties.hpp"
+#include "unit_test_utils/mocks/cpp_interfaces/interface/mock_icore.hpp"
+#include "unit_test_utils/mocks/cpp_interfaces/interface/mock_iexecutable_network_internal.hpp"
+#include "unit_test_utils/mocks/cpp_interfaces/interface/mock_iinference_plugin.hpp"
+
+using ::testing::_;
+using ::testing::MatcherCast;
+using ::testing::Matches;
+using ::testing::NiceMock;
+using ::testing::Return;
+using ::testing::StrEq;
+using ::testing::Throw;
+using Config = std::map<std::string, std::string>;
+using namespace MockMultiDevice;
+using ConfigParams = std::tuple<std::vector<std::string>>;
+
+// define a matcher to check if perf hint expects
+MATCHER_P(ComparePerfHint, perfHint, "Check if perf hint expects.") {
+    std::string arg_perfHint = "";
+    auto itor = arg.find(PluginConfigParams::KEY_PERFORMANCE_HINT);
+    if (itor != arg.end()) {
+        arg_perfHint = itor->second;
+    }
+
+    return perfHint == arg_perfHint;
+}
+
+class LoadNetworkWithCTPUTMockTest : public ::testing::TestWithParam<ConfigParams> {
+public:
+    std::shared_ptr<NiceMock<MockICore>> core;
+    std::shared_ptr<NiceMock<MockMultiPluginForLoadNetworkWithPropertiesTest>> plugin;
+    InferenceEngine::CNNNetwork simpleCnnNetwork;
```

Comment:
General comment: Why do you use legacy API? Can we migrate this test to OV 2.0?

Reply:
Because current interface between Core and AUTO plugin is legacy API. After OV 2.0 API enabled between Core and plugins, AUTO will migrate to OV 2.0 included test case.